Frontend/src/app/components/banner/CookieConsent.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+
+import { CookieIcon } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import { useEffect, useState } from "react";
+import { cn } from "@/lib/utils";
+
+import Link from "next/link";
+import { setCookiesAccepted } from "@/lib/cookies";
+import {
+  deleteSessionFromCookies,
+  commitSessionToCookies,
+} from "@/lib/session";
+
+export default function CookieConsent({
+  variant = "default",
+  onAcceptCallback = async () => {
+    await setCookiesAccepted();
+    await commitSessionToCookies();
+  },
+  onDeclineCallback = async () => {
+    try {
+      await deleteSessionFromCookies();
+      console.log("User denied cookie consent");
+    } catch (error) {
+      console.error("Error during handling deny:", error);
+    }
+  },
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [hide, setHide] = useState(false);
+
+  const accept = async () => {
+    setIsOpen(false);
+    document.cookie =
+      "cookieConsent=true; expires=Fri, 31 Dec 9999 23:59:59 GMT";
```

Comment:
We have an existing cookie to track whether the cookie was accepted- this is triggered by the `setCookiesAccepted();` you called earlier.  

Additionally, hard-coding an expiry value like this potentially opens us to issues. Please coordinate with the team if you plan on editing cookie logic in future like this.

Reply:
Deleted hard-coding, called setCookiesAccepted, commitSessionToCookies